src/WhitelistNFT.sol

```diff
@@ -31,4 +37,21 @@ contract WhitelistNFT is ERC721, Ownable {
         require(owner != address(0), "ZERO_ADDRESS");
         return super.balanceOf(owner);
     }
+
+
+    //@dev: _update is a function that is called by the internal transfer functions
+    function _update(
+        address to,
+        uint256 tokenId,
+        address auth
+    ) internal override returns (address) {
+        address owner = _ownerOf(tokenId);
+
+        // check if owner is zero address and to is not zero address
+        // this is to prevent transfer of soulbound NFT
+        // owner of NFT before minting is zero address
+        require(owner == address(0) && to != address(0), "Soulbound: Transfer failed");
```

Comment:
Gotcha, so I understand this correctly, this should ensure that "only mints can happen", no transfer, and no burning.

I think we should allow burning because if we decide to Airdrop this to people, and they don't want it, or do not want it in the future, they should have the right to burn!

Reply:
Good point. Let's allow burning